test/e2e/password_grant_check_test.go

```diff
@@ -32,3 +40,83 @@ func TestGitLabAsOIDCPasswordGrantCheck(t *testing.T) {
 
 	require.Equal(t, true, gitlabIDPConfig.UseAsChallenger, "gitlab is not configured as challenger")
 }
+
+func TestKeycloakAsOIDCPasswordGrantCheck(t *testing.T) {
+	testContext, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	kubeConfig := test.NewClientConfigForTest(t)
+
+	kubeClients, err := kubernetes.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	configClient, err := configv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	_, idpName, cleanups := test.AddKeycloakIDP(t, kubeConfig)
+	defer library.IDPCleanupWrapper(func() {
+		for _, c := range cleanups {
+			c()
+		}
+	})()
+
+	config, err := library.GrabOAuthServerConfig(kubeClients.CoreV1())
+	require.NoError(t, err)
+
+	kcIDPConfig := library.GetIDPByName(config, idpName)
+	require.NotNil(t, kcIDPConfig, "did not find idp %q in the config: %#v", idpName, config)
+
+	require.Equal(t, false, kcIDPConfig.UseAsChallenger, "keycloak is configured as challenger but it should not be")
+
+	oauthConfig, err := configClient.OAuths().Get(testContext, "cluster", v1.GetOptions{})
+	require.NoError(t, err)
+
+	var configIDP configv1.IdentityProvider
+	for _, idp := range oauthConfig.Spec.IdentityProviders {
+		if idp.Name == idpName {
+			configIDP = idp
+			break
+		}
+	}
+
+	// Get a keycloak client for the external KC URL
+	// skip TLS verify on the route's cert
```

Comment:
This might be a bit risky I think.

Keycloak generates it own, sef-signed cert if none is given (see [Keycloak container manual](https://github.com/keycloak/keycloak-containers/tree/master/server#setting-up-tlsssl)). The route pointing to Keycloak can support either Passthrough or Re-encrypt TLS Termination. So it's hard to tell what certificate (the one on the route or the one from Keycloak) you're ignoring here. 

Is there any way to supply a proper, OpenShift-generated certificate to Keycloak (see [how our Operator does it](https://github.com/keycloak/keycloak-operator/blob/master/pkg/model/keycloak_service.go#L21)) and use Re-Encrypt on the route? This way all certs should be fine.

Reply:
I can add a trust to the router certificate instead, although it did not seem necessary for the test.

Reply:
Fixed, it was actually simpler to do it that way